### PR TITLE
allow empty maxFeePerBlobGas for eth_call

### LINKED ISSUE #6709: make `maxBlobFeePerGas` optional for `eth_call` with blob `versionedHashes`
Follow up to #6657 

We shouldn’t require `maxFeePerBlobGas` for eth_call for the same reason regular gas fee settings aren’t necessary for eth_call for transactions without blobs.

Current implementation uses transaction validation so this would require checking the fields on the eth_call CallParameter

### PR DESCRIPTION
**Pull request by macfarla**
### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Most advanced CI tests are deferred until PR approval, but you could:

- [x] locally run all unit tests via: `./gradlew build`
- [x] locally run all acceptance tests via: `./gradlew acceptanceTest`
- [x] locally run all integration tests via: `./gradlew integrationTest`
- [x] locally run all reference tests via: `./gradlew ethereum:referenceTests:referenceTests`
- [x] locally run hive tests `engine-cancun` suite

## Fixed Issue(s)
fixes #6709

### CODE REVIEW

**Comment by macfarla** [comment]
@fab-10 does it makes sense to add the blob gas checks under the `isAllowExceedingBalance` flag - so don't check blobGasPrice or maxFeePerBlobGas if isAllowExceedingBalance is true

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java [hunk 1]
```diff
@@ -286,6 +287,8 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
             transactionValidationParams,
             operationTracer,
             blobGasPrice);
+    LOG.info("tx simulator " + transaction);
+    LOG.info("tx simulator " + result);
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:291]: remember to remove these

**Comment by fab-10** [comment]
> @fab-10 does it makes sense to add the blob gas checks under the `isAllowExceedingBalance` flag - so don't check blobGasPrice or maxFeePerBlobGas if isAllowExceedingBalance is true

yes it makes sense to me

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java (hunk 1 above)
Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java]: done

**Comment by macfarla** [comment]
> > @fab-10 does it makes sense to add the blob gas checks under the `isAllowExceedingBalance` flag - so don't check blobGasPrice or maxFeePerBlobGas if isAllowExceedingBalance is true
> 
> yes it makes sense to me

I have made this change. could you review @fab-10 - thanks!

**Comment by macfarla** [comment]
engine-cancun hive tests passing locally except for occasional known flakiness from `Invalid Missing Ancestor Syncing ReOrg, StateRoot, EmptyTxs=False, CanonicalReOrg=True, Invalid P9 (Cancun)`

**Review by fab-10 — APPROVED** [review]

**Pull request merged**